core/sr-primitives/src/traits.rs

```diff
@@ -768,6 +771,7 @@ pub trait ModuleDispatchError {
 }
 
 #[impl_for_tuples(1, 12)]
+#[allow(clippy::type_complexity)]
```

Comment:
@thiolliere for example, here - `clippy` flagged this module for several occurences of, in its words, "complex types" that can be refactored into smaller, simpler pieces.

From what I understand, this is a critical module that should not be modified carelessly, so that would be outside the scope of this PR, but just having this flag here means a simple `grep` in the future may help.